contracts/src/Utilities/README.md

```diff
@@ -0,0 +1,18 @@
+## Utility Contracts
+
+### ValidatorSetSig Contract
+
+This contract provides an alternative to traditional multi-signature contracts where instead of requiring signatures from `K` of `N` pre-specified signers, an aggregate signature is required from a quorum of the current validators a given blockchain.
+
+The contract leverages off-chain [Avalanche Warp Messages](https://docs.avax.network/build/cross-chain/awm/overview), which are manually approved for signing  by a chain's validators. It requires these messages to have the source address set to the zero address to enforce this.
+
+Note:
+1. The blockchain validating the message may or may not be the same chain where the target contract and the `ValidatorSetSig` contract are deployed.
+2. [Off-Chain Warp messages](https://github.com/ava-labs/subnet-evm/issues/729) are described in the linked issue. They are Warp messages that validators can include in their config to indicate that they are willing to sign them even though they are not a result of on-chain activity.
+
+#### Creating a valid Off-chain Warp Message for interaction with the ValidatorSetSig Contract
+
+1. ABI-encode a `ValidatorSetSigMessage` as defined in `ValidatorSetSig.sol`
+2. OPTIONAL: call `validateMessage` view method of the intended contract to confirm that the message has correct fields, including the nonce.
+3. Pack that as the payload of na [AddressedCall](https://github.com/ava-labs/avalanchego/blob/0c4efd743e1d737f4e8970d0e0ebf229ea44406c/vms/platformvm/warp/payload/addressed_call.go#L15) Warp Message format.
+4. Pack the `AddressedCall` as the payload of the [UnsignedWarpMessage](https://github.com/ava-labs/avalanchego/blob/f17ea6a7ab4036c41b693e47b94d8f0c81cb69ec/vms/platformvm/warp/unsigned_message.go#L14).
```

Comment:
should we reference aggregating signatures, and possibly point to awm-relayer here

Reply:
I don't think that's necessary since signature aggregation is separate from the validator signing the message provided via config

Reply:
I'm asking about including signature aggregation because the section `Creating a valid Off-chain Warp Message for interaction with the ValidatorSetSig Contract` is for "valid off-chain Warp message", and semantically we've used "valid Warp message" as one that includes valid aggregate signature

Reply:
would suggest removing "valid" from the header, but no strong pref

Reply:
Fair point, though I think the word "valid" should still be included. If we want to be technically accurate, what we're doing is providing a "valid unsigned off-chain Warp message". I don't think it's necessary to mention signature aggregation or any of the other details around actual message delivery here. Or if we do, keep it high level and say that signatures can be aggregated and packaged into a transaction in the same manner as any other Warp message.

Reply:
i agree high level is good enough